prime-router/docs/design/hl7-validation/design.md

```diff
@@ -0,0 +1,115 @@
+# Problem Statement
+As a sender/receiver I want submitted/received data to be validated to ensure data integrity and quality.
+
+There needs to exist a collection of settings and reusable components to perform varying types of validation at various points in the HL7 message's lifecycle.
+
+## Proposed Design
+![hl7-validation-annotated-architecture-diagram.png](annotated-architecture-diagram.png)
+
+We propose two validation "checkpoints":
+1. Sender validation
+2. Receiver Validation
```

Comment:
This is technically part of the FHIR validation design ticket. However, I may as well put this idea here, we should rewrite this document to be format independent and add sections for the various formats down below. Since, as Jimmy alludes to, the validation strategy and design is similar at a high-level independent of the format.

Reply:
So, instead of "Background Information/Dev Notes" the section should be titled "Validating HL7 messages" and then you can talk about HAPI library and whatever else. Feel free to add any thoughts you may have on this, anyone

Reply:
Just to follow up on this, we have decided to combine the two into one document, so hopefully the next proposal will have more than enough information on both.